galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/BackupCPS.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed Materials - Property of IBM
+ * 
+ * (c) Copyright IBM Corp. 2020.
+ */
+
+package dev.galasa.framework;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.osgi.service.component.annotations.Component;
+
+import dev.galasa.framework.FrameworkInitialisation;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IConfigurationPropertyStoreService;
+import dev.galasa.framework.spi.IFramework;
+
+@Component(service = { BackupCPS.class })
+public class BackupCPS {
+    
+    private Log  logger  =  LogFactory.getLog(this.getClass());
+    
+    private final boolean         file_append       = false;
+    private final boolean         autoFlush         = true;
+    
+    String charset = "UTF-8";
+    
+    private File                    file;
```

Comment:
The standard in Galasa is to use the java.nio.Files stuff rather than the old File,  sorry.

Reply:
This alteration should be reflected in the changes I'll be pushing shortly.